core/app/[locale]/(default)/account/[tab]/_components/wishlist-content/create-wishlist-form/_actions/create-wishlist.ts

```diff
@@ -0,0 +1,43 @@
+'use server';
+
+import { revalidatePath } from 'next/cache';
+import { z } from 'zod';
+
+import { createWishlist as createWishlistClient } from '~/client/mutations/create-wishlist';
+
+const CreateWishlistSchema = z.object({
+  name: z.string(),
+});
+
+export const createWishlist = async (formData: FormData) => {
+  const parsedData = CreateWishlistSchema.parse({
+    name: formData.get('name'),
+  });
+
+  const input = {
+    ...parsedData,
+    isPublic: true,
+  };
+
+  try {
+    const newWishlist = await createWishlistClient(input);
+
+    revalidatePath('/account/wishlists', 'page');
+
+    if (newWishlist) {
+      return {
+        status: 'success',
+        data: newWishlist,
+      };
+    }
```

Comment:
Curious about this behavior. Since we revalidate the path, do we ever use the result from data? 🤔

Reply:
We use this right after the mutation works to update state. Revalidate works slower, so using data from mutations makes interface more fast and responsive. The same for deleting wish list - we delete wish list from list right after successful mutation. But we still need retaliate for correct pagination work